misc/msvs/vcxproj.template

```diff
@@ -10,16 +10,16 @@
     <VCProjectUpgraderObjectName>NoUpgrade</VCProjectUpgraderObjectName>
   </PropertyGroup>
   %%PROPERTIES%%
-  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" Condition=" exists('$(VCTargetsPath)\Microsoft.Cpp.Default.props') "/>
   <PropertyGroup Label="Configuration">
     <ConfigurationType>Makefile</ConfigurationType>
     <UseOfMfc>false</UseOfMfc>
-    <PlatformToolset>v143</PlatformToolset>
```

Comment:
One of the goals of the generation is to make sure that all essential properties are set so that VS doesn't change the vcxproj automatically on open without user action.

Since you're moving the PlatformToolset property from the vcxproj to the individual platform files, what happens when you open the vcxproj on VS for a platform+target combo with no generated props file, does VS automatically add one, modifying the vcxproj in the process?

Reply:
VS does not add PlatformToolset, when switching to a solution configuration without it.